code/modules/admin/admin_ranks.dm

```diff
@@ -346,7 +349,7 @@ GLOBAL_PROTECT(admin_ranks)
 				D = new(R,adm_ckey)	//new admin
 
 			var/client/C = GLOB.directory[adm_ckey]	//find the client with the specified ckey (if they are logged in)
-			D.associate(C)						//link up with the client and add verbs
+			INVOKE_ASYNC(D, .proc/associate, C)						//link up with the client and add verbs

```

Comment:
i'm assuming by this point permissions have already been checked and i'm also assuming invoke_async won't pass usr into a proc?